flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/TableConfigUtils.java

```diff
@@ -90,6 +94,35 @@ public static CalciteConfig getCalciteConfig(TableConfig tableConfig) {
                 .orElse(CalciteConfig$.MODULE$.DEFAULT());
     }
 
+    /**
+     * Similar to {@link TableConfig#getLocalTimeZone()} but extracting it from a generic {@link
+     * ReadableConfig}.
+     *
+     * @see TableConfig#getLocalTimeZone()
+     */
+    public static ZoneId getLocalTimeZone(ReadableConfig tableConfig) {
+        String zone = tableConfig.get(TableConfigOptions.LOCAL_TIME_ZONE);
```

Comment:
I wonder if it makes sense to have the `ZoneId` as a type in the `ConfigOption` supported types...

Reply:
That would be nice, but it's a breaking change :(